harvester/periodic-jobs/src/main/java/dk/dbc/dataio/harvester/periodicjobs/FbiInfoConnector.java

```diff
@@ -29,19 +33,24 @@ public FbiInfoConnector(String baseUrl) {
         this.baseUrl = baseUrl;
     }
 
-    public boolean hasCover(RecordIdDTO recordId) {
-        HttpGet httpGet = new HttpGet(httpClient).withBaseUrl(baseUrl).withPathElements("manifestation", recordId.getAgencyId() + "-basis:" + recordId.getBibliographicRecordId()).withQueryParameter("trackingId", "uuid");
-        try {
-            FbiInfoResponse response = httpGet.executeAndExpect(FbiInfoResponse.class);
-            if (response == null || response.resources == null) return false;
-            return response.resources.contains("forside");
-        } catch (NotFoundException nfe) {
-            return false;
-        }
+    public Set<RecordIdDTO> hasCoverFilter(List<RecordIdDTO> recordIds) {
+        List<String> list = recordIds.stream().map(this::toManifestationId).toList();
+        HttpPost httpPost = new HttpPost(httpClient)
+                .withBaseUrl(baseUrl)
+                .withPathElements("manifestation", "forside")
+                .withJsonData(list);
+        //noinspection unchecked
+        List<String> ids = httpPost.executeAndExpect(List.class);
+        return ids.stream().map(this::toRecordIdDTO).collect(Collectors.toSet());
+    }
+
+    private String toManifestationId(RecordIdDTO recordIdDTO) {
+        return recordIdDTO.getAgencyId() + "-basis:" + recordIdDTO.getBibliographicRecordId();
```

Comment:
En pid vil typisk kun have -basis i sig hvis agencyId er 870970, for alle andre agencies er det vel -katalog ?

Reply:
Solr querien der starter balladen tager kun 870970 records